beacon_node/http_api/src/publish_attestations.rs

```diff
@@ -73,24 +75,57 @@ enum PublishAttestationResult {
 
 fn verify_and_publish_attestation<T: BeaconChainTypes>(
     chain: &Arc<BeaconChain<T>>,
-    attestation: &Attestation<T::EthSpec>,
+    attestation: &Either<Attestation<T::EthSpec>, SingleAttestation>,
     seen_timestamp: Duration,
     network_tx: &UnboundedSender<NetworkMessage<T::EthSpec>>,
     log: &Logger,
 ) -> Result<(), Error> {
+    let attestation = convert_to_attestation(chain, attestation)?;
     let attestation = chain
-        .verify_unaggregated_attestation_for_gossip(attestation, None)
+        .verify_unaggregated_attestation_for_gossip(&attestation, None)
         .map_err(Error::Validation)?;
 
-    // Publish.
-    network_tx
-        .send(NetworkMessage::Publish {
-            messages: vec![PubsubMessage::Attestation(Box::new((
-                attestation.subnet_id(),
-                attestation.attestation().clone_as_attestation(),
-            )))],
-        })
-        .map_err(|_| Error::Publication)?;
+    match attestation.attestation() {
+        types::AttestationRef::Base(_) => {
+            // Publish.
+            network_tx
+                .send(NetworkMessage::Publish {
+                    messages: vec![PubsubMessage::Attestation(Box::new((
+                        attestation.subnet_id(),
+                        attestation.attestation().clone_as_attestation(),
+                    )))],
+                })
+                .map_err(|_| Error::Publication)?;
+        }
+        types::AttestationRef::Electra(attn) => {
+            chain
```

Comment:
Can't we keep the `SingleAttestation` around prior to `convert_to_attestation` to prevent computing it all over again here?

Reply:
Yeah, agree. I think it's as easy as calling `attestation.single_attestation()` on the `VerifiedUnaggregatedAttestation`.

Reply:
Or yeah, just keeping the original `Either` and pattern-matching

Reply:
Optimised this here:

https://github.com/sigp/lighthouse/pull/6488/commits/8cc43c1a103db9959536b8b8806010f8be425668